src/overlays/actors/ovl_En_Peehat/z_en_peehat.h

```diff
@@ -4,11 +4,48 @@
 #include "ultra64.h"
 #include "global.h"
 
+#define GROUND_HOVER_HEIGHT 75.0f
+
+#define MAX_LARVA 3
+#define TYPE_GROUNDED -1
+#define TYPE_FLYING 0
+#define TYPE_LARVA 1
+
 struct EnPeehat;
 
+//params -1 = Ground Peahat
+//params 0 = Spawner Peahat
+//params 1 = Peahat larva
+
+typedef void (*EnPeehatActionFunc)(struct EnPeehat*, GlobalContext*);
+
 typedef struct EnPeehat {
     /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x2E0];
+    /* 0x014C */ SkelAnime skelAnime;
+    /* 0x0190 */ Vec3s jointTable[24];
+    /* 0x0220 */ Vec3s morphTable[24];
+    /* 0x02B0 */ s32 state;
+    /* 0x02B4 */ s32 isStateDieFirstUpdate;
+    /* 0x02B8 */ EnPeehatActionFunc actionFunc;
+    /* 0x02BC */ Vec3f bladeTip[2]; // used to simulate the peahat's blades "digging up" earth
+    /* 0x02D4 */ s32 unk2D4;
+    /* 0x02D8 */ f32 xzDistMax; //max xz dist to follow Link (TYPE_GROUNDED) or spawn larva (TYPE_FLYING)
```

Comment:
maybe maxFollowDist or something?

Reply:
follow doesn't make sense for TYPE_FLYING, since the way it works is it pops straight up and hovers in place.

Reply:
Just leave it then and maybe someone else will have a better name.